crypto/threads_none.c

```diff
@@ -203,56 +203,69 @@ int CRYPTO_THREAD_compare_id(CRYPTO_THREAD_ID a, CRYPTO_THREAD_ID b)
     return (a == b);
 }
 
-int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
+int CRYPTO_atomic_add(volatile int *val, int amount, volatile int *ret,
+                      CRYPTO_RWLOCK *lock)
```

Comment:
Surely adding `volatile` to all these isn't necessary?

Reply:
@tom-cosgrove-arm 
Certainly there are multiple ways to approach the warnings. A good API is unambiguous: easy to use correctly and hard to make mistakes.
1. All volatile: Pretty much all atomic functions for all platforms expect volatile parameters, the only exception being `__atomic_load`. Hence all wrappers should use volatile parameters to respect this. Arguments remain volatile all the way from declaration to use. This ensures synchronisation is done in RAM (as intended) and all variables remain consistent. To my understanding OpenSSL is a critical project so we should do things the right way and hint developers to declare parameters to atomic functions as volatile.
2. Minimal approach (many projects don't like if we touch more than necessary even if we fix something important): so use volatile only for those functions that have at least one call that passes a volatile variable. While this will fix the warnings, it will inevitably lead to confusion: why do some atomic wrappers take volatile parameters, while others don't? Makes no sense to me to make such a mess with the API.
3. Keep all functions as they are: non volatile. Then we can hide the warnings by intentionally casting away the volatile property, each time we pass such a variable. If all code is written correctly, all arguments to atomic functions should be atomic anyway, so we declare as volatile, then cast away to hide a warning, then internally it is promoted to volatile again. Makes no sense.